### PR TITLE
fix(core): fix alias creation bug

### PR DESCRIPTION
**Pull request by allegraharris**
Modifies the manner in which aliases are created. Three behaviours are changed in this pr: 

1) **Aliases generated without an explicit column name from a constant that contained one or more `.` were unintuitive and confusing.**
    a) These aliases were generated because aliases/column names cannot contain one or more `.`. To make column aliases more intuitive, we now rename such columns as `column1`, `column2`, etc.

_Old Behaviour:_
```sql
select '23.6.8.42', 36.8, 'questdb.io' 
``` 
| .8.42 _string_ | 8 _double_ | o _string_ |
|--------|--------|--------|
| 23.6.8.42 | 36.8 | questdb.io |

_New Behaviour:_
```sql
select '23.6.8.42', 36.8, 'questdb.io' 
``` 
| column1 _string_ | column2 _double_ | column3 _string_ |
|--------|--------|--------|
| 23.6.8.42 | 36.8 | questdb.io |

2) **Aliases were generated _while_ the select statement was being parsed, leading to the creation of duplicate column names unnecessarily which created an error**

_Old Behaviour:_
```sql
select 'a', 'a', 'a', 'a' a1 -> ERROR: Duplicate column [name=a1]
``` 

_New Behaviour:_
```sql
select 'a', 'a', 'a', 'a' a1
``` 
| a _char_ | a2 _char_ | a3 _char_ | a1 _char_ |
|--------|--------|--------|--------|
| a | a | a | a | 

3) **Aliases generated without an explicit column name from a numeric constant were unintuitive and confusing**

_Old Behaviour:_
```sql
select 1, 1, 1, 1, 1
``` 
| 1 _int_ | 11 _int_ | 12 _int_ | 13 _int_ | 14 _int_ |
|--------|--------|--------|--------|--------|
| 1 | 1 | 1 | 1 | 1 | 

_New Behaviour:_
```sql
select 1, 1, 1, 1, 1
``` 
| 1 _int_ | column _int_ | column1 _int_ | column2 _int_ | column3 _int_ |
|--------|--------|--------|--------|--------|
| 1 | 1 | 1 | 1 | 1 |

### CODE REVIEW

File: core/src/test/java/io/questdb/test/griffin/SqlParserTest.java [hunk 1]
```diff
@@ -7985,6 +8107,7 @@ public void testUndefinedBindVariables() throws SqlException {
     }
 
     @Test
+    @Ignore
```
Comment (ideoma) [inline core/src/test/java/io/questdb/test/griffin/SqlParserTest.java:8110]: please remove `@Ignore`

Comment (allegraharris) [reply core/src/test/java/io/questdb/test/griffin/SqlParserTest.java]: Removed in https://github.com/questdb/questdb/pull/3644/commits/d8eb986b70b3cb5cb585705d1253e027a316681c

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 48 / 49 (97.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|39|40|97.50%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 52 / 53 (98.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|39|40|97.50%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 52 / 53 (98.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|39|40|97.50%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 2]
```diff
@@ -1737,18 +1752,29 @@ private void parseSelectClause(GenericLexer lexer, QueryModel model) throws SqlE
...
             } else {
-                alias = createColumnAlias(expr, model);
+                alias = null;
+                col.setAlias(null);
+                columns.add(col);
             }
-            if (alias.length() == 0) {
-                throw err(lexer, null, "column alias cannot be a blank string");
+
+            if (alias != null) {
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:1760]: nit: better to concat two ifs which check alias for null

```java
            if (alias != null) {
                if (alias.length() == 0) {
                    throw err(lexer, null, "column alias cannot be a blank string");
                }
                col.setAlias(alias);
                model.addBottomUpColumn(colPosition, col, false);
            } else { //only add to bottom up columns / bottom up column names which are responsible for ordering, the others will be updated later
                model.addFieldForNullAlias(col);
            }
```

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 3]
```diff
@@ -174,6 +179,14 @@ private CharSequence createColumnAlias(ExpressionNode node, QueryModel model) {
         );
     }
 
+    private CharSequence createConstColumnAlias(LowerCaseCharSequenceObjHashMap<QueryColumn> aliasToColumnMap) {
+        while (aliasToColumnMap.contains(column + Integer.toString(digit))) {
```
[the remark this answers is not in the document]
Comment (allegraharris) [reply core/src/main/java/io/questdb/griffin/SqlParser.java]: Fixed in https://github.com/questdb/questdb/pull/3644/commits/7f2e3eb29a270738258d18362b6b46572d39d475

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 4]
```diff
@@ -1773,6 +1799,19 @@ private void parseSelectClause(GenericLexer lexer, QueryModel model) throws SqlE
                 throw err(lexer, tok, "',', 'from' or 'over' expected");
             }
         }
+
+        CharSequence alias;
+
+        for (int i = 0; i < columns.size(); i++) {
+            CharSequence token = columns.get(i).getAst().token;
+            if (columns.get(i).getAst().type == ExpressionNode.CONSTANT && Chars.indexOf(columns.get(i).getAst().token, '.') != -1) {
```
[the remark this answers is not in the document]
Comment (allegraharris) [reply core/src/main/java/io/questdb/griffin/SqlParser.java]: Fixed in https://github.com/questdb/questdb/pull/3644/commits/f94f72e61f576e7a38758709943ee6a0ebb9c069

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 5]
```diff
@@ -2193,6 +2232,23 @@ private int toColumnType(GenericLexer lexer, CharSequence tok) throws SqlExcepti
         return tok;
     }
 
+    private void updateMapsAndLists(QueryModel model, CharSequence alias, CharSequence token, int i) {
+        QueryColumn nullColumn = columns.get(i);
```
[the remark this answers is not in the document]
Comment (allegraharris) [reply core/src/main/java/io/questdb/griffin/SqlParser.java]: Fixed in https://github.com/questdb/questdb/pull/3644/commits/f94f72e61f576e7a38758709943ee6a0ebb9c069

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 6]
```diff
@@ -1773,6 +1799,19 @@ private void parseSelectClause(GenericLexer lexer, QueryModel model) throws SqlE
                 throw err(lexer, tok, "',', 'from' or 'over' expected");
             }
         }
+
+        CharSequence alias;
```
[the remark this answers is not in the document]
Comment (allegraharris) [reply core/src/main/java/io/questdb/griffin/SqlParser.java]: Fixed in https://github.com/questdb/questdb/pull/3644/commits/975c7ec812494c0684e780f2a199e6506d97c11a

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 2 above)
Comment (allegraharris) [reply core/src/main/java/io/questdb/griffin/SqlParser.java]: Fixed in https://github.com/questdb/questdb/pull/3644/commits/b06bfa942018ee4f648f824a8e90485f624eb078

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 56 / 57 (98.25%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|43|44|97.73%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlUtil.java|2|2|100.00%|

**Pull request merged**